ddtrace/tracer/telemetry.go

```diff
@@ -108,9 +110,22 @@ func startTelemetry(c *config) {
 				Value: fmt.Sprintf("rate:%f_maxPerSecond:%f", rule.Rate, rule.MaxPerSecond)})
 	}
 	if c.orchestrionCfg.Enabled {
+		tags := make([]string, 0, len(c.orchestrionCfg.Metadata))
 		for k, v := range c.orchestrionCfg.Metadata {
 			telemetryConfigs = append(telemetryConfigs, telemetry.Configuration{Name: "orchestrion_" + k, Value: v})
+			tags = append(tags, k+":"+v)
 		}
+		if testing.Testing() {
+			// In tests, ensure tags are consistently ordered...
+			slices.Sort(tags)
+		}
```

Comment:
There is another PR for removing `testing` from `dd-trace-go`: #3017.

You can achieve the same effect if you move the `slices.Sort(tags)` into `internal/telemetry/telemetrytest.MockClient#Record` like this:

```go
// Record stores the value for the given metric. It is currently mocked for `Gauge` and `Distribution` metric types.
func (c *MockClient) Record(ns telemetry.Namespace, kind telemetry.MetricKind, name string, val float64, tags []string, common bool) {
	slices.Sort(tags)
	c.On("Gauge", ns, kind, name, val, tags, common).Return()
	c.On("Record", ns, kind, name, val, tags, common).Return()
	_ = c.Called(ns, kind, name, val, tags, common)
```

This should be a safe change - I didn't run the whole CI - because that function has a bug. The calls to `MockClient#On` are ignoring the kind argument, thus being unable to match any call before applying this suggested fix.